packages/datadog-plugin-prisma/test/index.spec.js

```diff
@@ -0,0 +1,310 @@
+  }
+
+  model User {
+  id    Int     @id @default(autoincrement())
+  email String  @unique
+  name  String?
+}
+    `
+  await fs.writeFile(generatedSchemaPath, schema) /
+  execSync(`${prismaBin} generate --schema="${generatedSchemaPath}"`, {
+    cwd: tempDir, // Ensure the current working directory is where the schema is located
+    stdio: 'inherit'
+  })
+  return tempDir
+}
+// Cleanup function to remove the mock Prisma client and its dependencies
+// from the temporary directory after tests are done
+async function cleanupMockPrismaClient (directory) {
+  const nodeModulesPath = path.join(directory, 'node_modules')
+  const packageJsonPath = path.join(directory, 'package.json')
+  const prismaSchemaPath = path.join(directory, 'schema.prisma')
+
+  await fs.rm(nodeModulesPath, { recursive: true, force: true })
+  await fs.rm(packageJsonPath, { force: true })
+
+  await fs.rm(prismaSchemaPath, { force: true })
+}
+
+describe('Plugin', () => {
+  let prisma
+  let prismaClient
+  let tracingHelper
+
+  describe('prisma', () => {
+    let tempDir
+
+    after(async () => {
+      await cleanupMockPrismaClient(tempDir)
+    })
+    withVersions('prisma', ['@prisma/client'], version => {
+      before(async () => {
+        tempDir = await generatePrismaClient(version)
+      })
+
+      describe('without configuration', () => {
+        before(() => {
+          return agent.load('prisma')
+        })
+
+        beforeEach(() => {
+          prisma = require(`../../../versions/@prisma/client@${version}`).get()
+          prismaClient = new prisma.PrismaClient()
+          tracingHelper = global.PRISMA_INSTRUMENTATION.helper || global.V0_PRISMA_INSTRUMENTATION.helper
+        })
+
+        after(() => { return agent.close({ ritmReset: false }) })
+
+        it('should do automatic instrumentation', done => {
+          agent.use(traces => {
+            expect(traces[0][0].resource).to.equal('queryRaw')
+            expect(traces[0][0].meta).to.have.property('prisma.type', 'client')
+            expect(traces[0][0].meta).to.have.property('prisma.method', 'queryRaw')
+            expect(traces[0][0]).to.have.property('name', expectedSchema.client.opName)
+            expect(traces[0][0]).to.have.property('service', expectedSchema.client.serviceName)
+
+            // grabbing actual db query span
+            const engineDBSpan = traces[0].find(span => span.meta['prisma.name'] === 'db_query')
+            expect(engineDBSpan).to.have.property('resource', 'SELECT 1')
+            expect(engineDBSpan).to.have.property('type', 'sql')
+            expect(engineDBSpan.meta).to.have.property('span.kind', 'client')
+            expect(engineDBSpan).to.have.property('name', expectedSchema.engine.opName)
+            expect(engineDBSpan).to.have.property('service', expectedSchema.engine.serviceName)
+          }).then(done).catch(done)
+
+          prismaClient.$queryRaw`SELECT 1`.catch(done)
```

Comment:
I am currently trying to make use of more async/await in the tests for readability reasons and because we often see "done called more than once", which would be fixed with this:

```suggestion
          })

          await Promise.all([
            prismaClient.$queryRaw`SELECT 1`,
            tracingPromise
          ])
```

It does need the method to be changed to an async one and the `tracingPromise` must be defined.

I added this comment as example for the rest of the tests.